### PR TITLE
FEATURE: `SporkID` and `ProtocolVersion`

### PR DESCRIPTION
**Pull request by jordanschalm**
* #786 
* #806 
* https://github.com/onflow/flow-go/pull/1474

### CODE REVIEW

**Review by smnzhu — APPROVED** [review]

File: state/protocol/inmem/convert.go [hunk 1]
```diff
@@ -189,6 +220,18 @@ func ClusterFromEncodable(enc EncodableCluster) (*Cluster, error) {
 // root bootstrap state. This is used to bootstrap the protocol state for
 // genesis or post-spork states.
 func SnapshotFromBootstrapState(root *flow.Block, result *flow.ExecutionResult, seal *flow.Seal, qc *flow.QuorumCertificate) (*Snapshot, error) {
+	return SnapshotFromBootstrapStateWithProtocolVersion(root, result, seal, qc, flow.DefaultProtocolVersion)
```
Comment (kc1116) [inline state/protocol/inmem/convert.go:223]: When bootstrapping from root state will the protocol version always be flow.DefaultProtocolVersion ?

**Review by kc1116 — APPROVED** [review]

File: state/protocol/inmem/convert.go (hunk 1 above)
Comment (jordanschalm) [reply state/protocol/inmem/convert.go:223]: For real networks it should be set during sporking, as it's a [required flag](https://github.com/onflow/flow-go/pull/810/files#diff-b886924fe107068c99c4240a7fabc7f095cdf575f60a971e7041bf2ba77494e8R106)

**Comment by jordanschalm** [comment]
bors merge

**Comment by jordanschalm** [comment]
bors merge

**Pull request merged**